### PR TITLE
LPS-142497 Use interval if ResizeObserver is not supported

### PR DESCRIPTION
**Pull request by p2kmgcl**

### CODE REVIEW

**Comment by p2kmgcl** [comment]
ci:test:sf

**Comment by p2kmgcl** [comment]
ci:test:relevant

**Comment by p2kmgcl** [comment]
ci:test:relevant

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/topper/TopperLabel.js [hunk 1]
```diff
@@ -12,59 +12,166 @@
...
+				(entries) => {
+					entries.forEach((entry) => {
+						if (entry.target === itemElement) {
+							const boundingClientRect = itemElement.getBoundingClientRect();
+							const computedStyle = globalContext.window.getComputedStyle(
+								itemElement
+							);
+
+							itemElementMarginRight =
+								(computedStyle.marginRight.endsWith('px') &&
```
Comment (victorg1991) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/topper/TopperLabel.js]: can't we asume here that computedStyle for margin is always returning pixels unit?

Comment (p2kmgcl) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/topper/TopperLabel.js]: :thinking: I thought that this function returned the computed value as written by users (ex. "2rem"), but it looks like I am wrong. Will update it tomorrow :smile: 

Thanks!

**Comment by liulukiwi** [comment]
ci:test:bundle

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/topper/TopperLabel.js (hunk 1 above)
Comment (p2kmgcl) [reply modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/topper/TopperLabel.js]: Updated :D

**Comment by liulukiwi** [comment]
ci:test:relevant

**Comment by Tim-Cao** [comment]
Just start reviewing

**Comment by Tim-Cao** [comment]
Hi @p2kmgcl, it's hard to distinguish some fragments in page editor after move the topper bar to the page canvas.
For example, Content Display, Content Flags, Collection Filter, Applied Filters, Content Ratings, Menu Display.
Is that possible to add a hidden element for title in the same level with **page-editor__topper__content**?
For example,
image

**Comment by p2kmgcl** [comment]
Good point, I think we can add an aria-label attribute to each element. Let me push a commit to see if it's enough.

**Comment by p2kmgcl** [comment]
@Tim-Cao there it is https://github.com/liferay-echo/liferay-portal/pull/7124/commits/9c7b82a017ff42ec5cfcbf11b3df661cd6c58c67

Now every fragment has an "aria-label" property with the name, and a "aria-labelledby" property with the Topper label ID if it is active.

**Comment by Tim-Cao** [comment]
Really nice! Thanks @p2kmgcl

**Comment by Tim-Cao** [comment]
See https://github.com/liferay-echo/liferay-portal/pull/7168

**Comment by Tim-Cao** [comment]
ci:close

**Pull request closed without merging**